client/network/sync/src/engine.rs

```diff
@@ -799,94 +805,110 @@ where
 	/// Called by peer when it is disconnecting.
 	///
 	/// Returns a result if the handshake of this peer was indeed accepted.
-	pub fn on_sync_peer_disconnected(&mut self, peer: PeerId) -> Result<(), ()> {
-		if self.peers.remove(&peer).is_some() {
-			if self.important_peers.contains(&peer) {
-				log::warn!(target: "sync", "Reserved peer {} disconnected", peer);
-			} else {
-				log::debug!(target: "sync", "{} disconnected", peer);
-			}
+	pub fn on_sync_peer_disconnected(&mut self, peer: PeerId) {
+		if !self.peers.remove(&peer).is_some() {
+			log::debug!(target: LOG_TARGET, "{peer} does not exist in `SyncingEngine`");
```

Comment:
I've seen this error trigger in different from ("Disconnected peer who was previously refused by the syncing layer") so there is definitely some race in the code which actually triggers so I'm a little hesitant to adding `debug_assert!()` because I don't know how common the case is and how much it might hinder, e.g., Versi tests which are run with debug asserts. But I'll convert it to an error